controllers/datasource_controller.go

```diff
@@ -156,7 +156,7 @@
 	}
 
 	// check datasource
-	if err := ds.Check(ctx, nil); err != nil {
+	if err := ds.Check(ctx, instance.Spec.OSS.DeepCopy()); err != nil {
 		return r.UpdateStatus(ctx, instance, err)
 	}
```

Comment:
**Suggestion:** Consider adding a log message before returning the error in the `Checkdatasource` function.
```suggestion
 func (r *DatasourceReconciler) Checkdatasource(ctx context.Context, logger logr.Logger, instance *v1alpha1.Datasource) error {
     // check datasource
     if err := ds.Check(ctx, instance.Spec.OSS.DeepCopy()); err != nil {
         logger.Error(err, "Failed to check datasource")
         return r.UpdateStatus(ctx, instance, err)
     }
 }
```